tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -858,44 +858,82 @@ void test_variables_slice() noexcept {
+  fill_with_random_values(make_not_null(&slice1_vals), make_not_null(&gen),
+                          make_not_null(&dist));
+
+  using Tensor =
+      typename VariablesTestTags_detail::tensor_tag<VectorType>::type;
   const Index<2> extents{{{4, 2}}};
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> vars(
+  Variables<tmpl::list<VariablesTestTags_detail::tensor_tag<VectorType>>> vars(
       extents.product());
-  get<VariablesTestTags_detail::vector>(vars) =
-      Vector{{{{1110000., 1120000., 1130000., 1140000., 1210000., 1220000.,
-                1230000., 1240000.},
-               {2110000., 2120000., 2130000., 2140000., 2210000., 2220000.,
-                2230000., 2240000.},
-               {3110000., 3120000., 3130000., 3140000., 3210000., 3220000.,
-                3230000., 3240000.}}}};
-
+  get<VariablesTestTags_detail::tensor_tag<VectorType>>(vars) =
+      Tensor{{{orig_vals.at(0), orig_vals.at(1), orig_vals.at(2)}}};
   {
     const auto slice_extents = extents.slice_away(0);
-    Variables<tmpl::list<VariablesTestTags_detail::vector>> slice(
-        slice_extents.product(), 0.);
-    get<VariablesTestTags_detail::vector>(slice) =
-        Vector{{{{1100., 1200.}, {2100., 2200.}, {3100., 3200.}}}};
+    Variables<tmpl::list<VariablesTestTags_detail::tensor_tag<VectorType>>>
+        slice(slice_extents.product(), 0.);
+    get<VariablesTestTags_detail::tensor_tag<VectorType>>(slice) =
+        Tensor{{{slice1_vals.at(0), slice1_vals.at(1), slice1_vals.at(2)}}};
     add_slice_to_data(make_not_null(&vars), slice, extents, 0, 2);
   }
+
   {
     const auto slice_extents = extents.slice_away(1);
-    Variables<tmpl::list<VariablesTestTags_detail::vector>> slice(
-        slice_extents.product(), 0.);
-    get<VariablesTestTags_detail::vector>(slice) = Vector{
-        {{{11., 12., 13., 14.}, {21., 22., 23., 24.}, {31., 32., 33., 34.}}}};
+    Variables<tmpl::list<VariablesTestTags_detail::tensor_tag<VectorType>>>
+        slice(slice_extents.product(), 0.);
+    get<VariablesTestTags_detail::tensor_tag<VectorType>>(slice) =
+        Tensor{{{slice0_vals.at(0), slice0_vals.at(1), slice0_vals.at(2)}}};
     add_slice_to_data(make_not_null(&vars), slice, extents, 1, 1);
   }
 
-  CHECK((Vector{{{{1110000., 1120000., 1131100., 1140000., 1210011., 1220012.,
-                   1231213., 1240014.},
-                  {2110000., 2120000., 2132100., 2140000., 2210021., 2220022.,
-                   2232223., 2240024.},
-                  {3110000., 3120000., 3133100., 3140000., 3210031., 3220032.,
-                   3233233., 3240034.}}}}) ==
-        get<VariablesTestTags_detail::vector>(vars));
+  // The slice0_vals should have been added to the second half of each of the
+  // three vectors in the tensor. The slice1_vals should have been added to
+  // entries 2 and 6 in each vector.
+  auto expected = Tensor{
+      {{{orig_vals.at(0).at(0), orig_vals.at(0).at(1),
```

Comment:
manually format this for readability. and use `// clang-format off` and `// clang-format on`